File: hooks/atum-session-start.py
#!/usr/bin/env python3
"""
ATUM Audit -- SessionStart hook for Claude Code.

Auto-detects and initializes ATUM when Claude Code starts in a project.
Outputs JSON additionalContext with ATUM status.

Hook type: SessionStart
Exit code: always 0 (never blocks Claude)
"""

import json
import os
import sys


def check_hook_integrity():
    """Verify critical safety hooks are accessible."""
    hooks_dir = os.path.join(os.path.expanduser("~"), ".claude", "hooks")
    critical = ["file-guard.js", "secret-scanner.js", "git-guard.js",
                "anti-rationalization.js", "pre-completion-gate.js"]
    missing = [h for h in critical if not os.path.isfile(os.path.join(hooks_dir, h))]
    if missing:
        print(f"CRITICAL: Missing safety hooks: {', '.join(missing)}", file=sys.stderr)
        print(f"ATUM safety compromised. Run install.sh to repair.", file=sys.stderr)


def main():
    check_hook_integrity()

    # Add ATUM library to path
    lib_dir = os.environ.get("ATUM_PROJECT_DIR", "")
    if lib_dir and lib_dir not in sys.path:
        sys.path.insert(0, lib_dir)

    from atum_audit.discovery import find_config, find_project_root, auto_init_project

    # Detect current working directory
    cwd = os.environ.get("PWD", os.getcwd())

    # Check if already an ATUM project
    config_path = find_config(cwd)
    if config_path is not None:
        project_name = config_path.parent.name
        context = f"[ATUM] Tracking project: {project_name} ({config_path.parent})"
    else:
        # Try to find a project root and auto-init
        project_root = find_project_root(cwd)
        if project_root is not None:
            config_path = auto_init_project(project_root, lib_dir=lib_dir or None)
            project_name = project_root.name
            context = f"[ATUM] Auto-initialized in: {project_name} ({project_root})"
        else:
            # Not a project directory -- nothing to do
            context = ""

    if context:
        output = {
            "hookSpecificOutput": {
                "hookEventName": "SessionStart",
                "additionalContext": context,
            }
        }
        print(json.dumps(output))


if __name__ == "__main__":
    try:
        main()
    except ImportError as e:
        print(f"[ATUM] WARNING: import failed — {e}. Check ATUM_PROJECT_DIR.", file=sys.stderr)
    except Exception as e:
        print(f"[ATUM] WARNING: session-start hook error — {e}", file=sys.stderr)
    sys.exit(0)

File: hooks/atum-session-start.js
#!/usr/bin/env node
/**
 * ATUM Audit — SessionStart hook (JS wrapper)
 *
 * Fast JS entry point that:
 *  1. Checks hook integrity (JS, ~1ms)
 *  2. Delegates to Python only if ATUM_PROJECT_DIR is set and atum_audit is available
 *
 * This avoids paying ~200ms Python startup when atum_audit isn't installed.
 * Hook type: SessionStart. Exit: always 0.
 */
'use strict';

const fs = require('fs');
const path = require('path');
const { execFileSync } = require('child_process');

function checkHookIntegrity() {
  const hooksDir = __dirname;
  const critical = ['file-guard.js', 'secret-scanner.js', 'git-guard.js',
                    'anti-rationalization.js', 'pre-completion-gate.js'];
  const missing = critical.filter(h => !fs.existsSync(path.join(hooksDir, h)));
  if (missing.length > 0) {
    process.stderr.write(`WARNING: Missing safety hooks: ${missing.join(', ')}\n`);
  }
}

// Drain stdin
let stdinData = '';
process.stdin.setEncoding('utf8');
process.stdin.on('data', chunk => { stdinData += chunk; });
process.stdin.on('end', () => {
  try {
    checkHookIntegrity();

    const libDir = process.env.ATUM_PROJECT_DIR || '';
    if (!libDir) {
      // No ATUM project configured — skip Python call entirely
      process.exit(0);
    }

    // Delegate to Python script (only when atum_audit is available)
    const pyScript = path.join(__dirname, 'atum-session-start.py');
    if (!fs.existsSync(pyScript)) process.exit(0);

    try {
      const result = execFileSync('python3', [pyScript], {
        input: stdinData,
        encoding: 'utf8',
        timeout: 5000,
        env: { ...process.env, ATUM_PROJECT_DIR: libDir },
      });
      if (result.trim()) process.stdout.write(result);
    } catch {
      // Python not available or atum_audit not installed — silent skip
    }
  } catch {
    // Never block
  }
  process.exit(0);
});
